File: java/src/test/java/mx/iteso/CommentTest.java
package mx.iteso;
import static org.junit.Assert.*;
import static org.mockito.Mockito.*;

import org.junit.Before;
import org.junit.Test;

import mx.iteso.User;
import mx.iteso.privileges.Approve;
import mx.iteso.privileges.Comment;
 

public class CommentTest {
    private User user;
    @Before
    public void setUp(){
        user = mock((User.class));
    }
    @Test
    public void testComment(){
        User com = new Comment(user);
        when(user.canComment()).thenReturn(false);
        boolean canComment = com.canComment();
        assertTrue(canComment);
    }
    @Test
    public void testDescription(){
        User com = new Comment(user);
        when(user.getDesc()).thenReturn("User with privileges: ");
        String desc = com.getDesc();
        assertEquals("User with privileges:  comment ", desc);
    }
}

File: python/src/unittest/python/CommentTest.py
from mockito import mock, verify, when
from User import *
from Comment import *
import unittest


class CommentTest(unittest.TestCase):
    def setUp(self):
        self.user = mock(User, strict=False)

    def test_CommentPermission(self):
        commenter = Comment(self.user)
        when(self.user).commentPermission().thenReturn(False)
        commentAdded = commenter.commentPermission()
        self.assertTrue(commentAdded)

    def test_Description(self):
        commenter = Comment(self.user)
        when(self.user).getDescription().thenReturn("Current user")
        desc = commenter.getDescription()
        self.assertEquals("Comment: Current user", desc)
